dissect/target/target.py

```diff
@@ -632,6 +632,14 @@ def has_function(self, function: str) -> bool:
         except PluginError:
             return False
 
+    def os_plugin_class_name(self) -> str:
```

Comment:
It is meant to be called from outside the object (plugin_function_index), so making it private does not make sense to me? There needs to be a way to determine whether a target has an os plugin associated with it and if so, which one that is. Otherwise it is difficult to match the correct OS plugin, given the fact that they have no compatible_check() method.

Reply:
I usually consider these kinds of things to be "project internal" rather than "object internal". There's really no reason to call that method outside of any of the internal code in `dissect.target`.

Reply:
Okay, well if that is the case, am I allowed to access _os_plugin from the outside directly then?

Reply:
I don't see why not. This is all internal code of `dissect.target`, it should be allowed to access `_` methods and attributes.

Reply:
Okay, well in a "traditional" sense of OOP, this was not allowed of course, but I am happy with your "project internal" definition of OO encapsulation! (:smiley:)

Reply:
Yeah, but then we'd have a `InternalTarget` object and `getInternalState` methods ;) I prefer to bend the rules a little bit here for a bit more readable code.